pkg/checks/health.go

```diff
@@ -122,23 +135,30 @@ func (h *Health) Shutdown(_ context.Context) error {
 
 // SetConfig sets the configuration for the health check
 func (h *Health) SetConfig(_ context.Context, config any) error {
-	var checkCfg HealthConfig
-	if err := mapstructure.Decode(config, &checkCfg); err != nil {
+	var c HealthConfig
+	if err := mapstructure.Decode(config, &c); err != nil {
 		return ErrInvalidConfig
 	}
-	h.config = checkCfg
+	c.Interval *= time.Second
+	c.Retry.Delay *= time.Second
+	h.mu.Lock()
+	defer h.mu.Unlock()
+	h.config = c
+
 	return nil
 }
 
 // SetClient sets the http client for the health check
-func (h *Health) SetClient(_ *http.Client) {
-	// TODO: implement with issue #31
+func (h *Health) SetClient(c *http.Client) {
+	h.mu.Lock()
+	defer h.mu.Unlock()
+	h.client = c
 }
 
 // Schema provides the schema of the data that will be provided
-// by the heath check
+// by the health check
 func (h *Health) Schema() (*openapi3.SchemaRef, error) {
-	return OpenapiFromPerfData[healthData](healthData{})
+	return OpenapiFromPerfData[[]HealthResult]([]HealthResult{})
```

Comment:
The current implementation of the health check would change the API schema of the check. 

Should we keep the old schema?

Reply:
I think it's fine if we change this

Reply:
@NiklasTreml Generally what schema would you prefer?
```json
{
  "data": {
    "https://www.google.com/": "healthy",
    "https://www.example.com/": "healthy"
  },
  "timestamp": "2024-01-11T15:19:15.315158452+01:00",
  "error": ""
}
```
or
```json
{
  "data": [
    {
      "target": "https://www.google.com/",
      "status": "healthy"
    },
    {
      "target": "https://www.example.com/",
      "status": "healthy"
    }
  ],
  "timestamp": "2024-01-11T15:19:15.315158452+01:00",
  "error": ""
}
```
The current implementation is like the second schema but I think I'd prefer the first.

Reply:
First one is fine by me